internal/clients/api_client.go

```diff
@@ -205,80 +91,19 @@ func NewAcceptanceTestingClient() (*ApiClient, error) {
-	if esConn, ok := d.GetOk("elasticsearch_connection"); ok {
-		config := elasticsearch.Config{}
-		config.Header = http.Header{"User-Agent": []string{fmt.Sprintf("elasticstack-terraform-provider/%s", defaultClient.version)}}
 
-		// there is always only 1 connection per resource
-		conn := esConn.([]interface{})[0].(map[string]interface{})
+	if _, ok := d.GetOk(esConnectionKey); ok {
+		apiClient, diags := newEsApiClient(d, esConnectionKey, defaultClient.version)
 
-		if u := conn["username"]; u != nil {
-			config.Username = u.(string)
-		}
-		if p := conn["password"]; p != nil {
-			config.Password = p.(string)
-		}
-		if k := conn["api_key"]; k != nil {
-			config.APIKey = k.(string)
-		}
-		if endpoints := conn["endpoints"]; endpoints != nil {
-			var addrs []string
-			for _, e := range endpoints.([]interface{}) {
-				addrs = append(addrs, e.(string))
-			}
-			config.Addresses = addrs
-		}
-		if insecure := conn["insecure"]; insecure.(bool) {
-			tlsClientConfig := ensureTLSClientConfig(&config)
-			tlsClientConfig.InsecureSkipVerify = true
-		}
-		if caFile, ok := conn["ca_file"]; ok && caFile.(string) != "" {
-			caCert, err := os.ReadFile(caFile.(string))
-			if err != nil {
-				return nil, fmt.Errorf("Unable to read ca_file: %w", err)
-			}
-			config.CACert = caCert
-		}
-		if caData, ok := conn["ca_data"]; ok && caData.(string) != "" {
-			config.CACert = []byte(caData.(string))
+		if diags.HasError() {
+			return nil, diags
 		}
 
-		if certFile, ok := conn["cert_file"]; ok && certFile.(string) != "" {
-			if keyFile, ok := conn["key_file"]; ok && keyFile.(string) != "" {
-				cert, err := tls.LoadX509KeyPair(certFile.(string), keyFile.(string))
-				if err != nil {
-					return nil, fmt.Errorf("Unable to read certificate or key file: %w", err)
-				}
-				tlsClientConfig := ensureTLSClientConfig(&config)
-				tlsClientConfig.Certificates = []tls.Certificate{cert}
-			} else {
-				return nil, fmt.Errorf("Unable to read key file: Path to key file has not been configured or is empty")
-			}
-		}
-		if certData, ok := conn["cert_data"]; ok && certData.(string) != "" {
-			if keyData, ok := conn["key_data"]; ok && keyData.(string) != "" {
-				cert, err := tls.X509KeyPair([]byte(certData.(string)), []byte(keyData.(string)))
-				if err != nil {
-					return nil, fmt.Errorf("Unable to parse certificate or key: %w", err)
-				}
-				tlsClientConfig := ensureTLSClientConfig(&config)
-				tlsClientConfig.Certificates = []tls.Certificate{cert}
-			} else {
-				return nil, fmt.Errorf("Unable to parse key: Key data has not been configured or is empty")
-			}
-		}
-
-		es, err := elasticsearch.NewClient(config)
-		if err != nil {
-			return nil, fmt.Errorf("Unable to create Elasticsearch client")
-		}
-		if logging.IsDebugOrHigher() {
-			es.Transport = newDebugTransport("elasticsearch", es.Transport)
-		}
-		return &ApiClient{es, defaultClient.version}, nil
+		return apiClient, nil
 	} else { // or return the default client
```

Comment:
It never made sense for this to be in an else block. This fn should just be

```
defaultClient := ...
if esConn exists {
  return newApiClient(...)
}

return defaultClient, nil
```

Reply:
I had it cleaner like this initially, but the acceptance tests were failing without the condition for `elasticsearch_connection` (it worked without it for the case from `elasticsearch`).

I'll try again to verify that was the case and not just the state I had it in at the time.